### PR TITLE
support compression

### PR DESCRIPTION
**Pull request by YangKian**

### CODE REVIEW

File: hstream.proto [hunk 1]
```diff
@@ -211,7 +211,20 @@ message Stream {
...
+  Gzip = 1;
+}
+
+message BatchedRecord {
+  CompressionType compressionType = 1;
+  // Required.
+  // The time at which the message was published,
+  // populated by the server.
+  google.protobuf.Timestamp publish_time = 2;
+  uint64 batchSize = 3;
```
Comment (4eUeP) [inline hstream.proto:225]: What does this `batchSize` do? Can you make some comments about this field?

Comment (YangKian) [reply hstream.proto:225]: explained in the design document, will add a comment here too

Comment (YangKian) [reply hstream.proto:230]: fixed

File: hstream.proto [hunk 2]
```diff
@@ -211,7 +211,25 @@ message Stream {
...
+  None = 0;
+  Gzip = 1;
+}
+
+message BatchedRecord {
+  CompressionType compressionType = 1;
+  // Required.
+  // The time at which the message was published,
+  // populated by the server.
+  google.protobuf.Timestamp publish_time = 2;
```
Comment (daleiz) [inline hstream.proto:224]: publishTime

**Review by daleiz — APPROVED** [review]

**Pull request merged**